src/nvim/terminal.c

```diff
@@ -248,21 +248,21 @@ static int parse_osc8(VTermStringFragment frag, int *attr)
   }
 
   // Find the terminator
-  const size_t start = i;
+  const int start = i;
   for (; i < frag.len; i++) {
     if (frag.str[i] == '\a' || frag.str[i] == '\x1b') {
       break;
     }
   }
 
-  const size_t len = i - start;
+  const int len = i - start;
   if (len == 0) {
     // Empty OSC 8, no URL
     *attr = 0;
     return 1;
   }
 
-  char *url = xmemdupz(&frag.str[start], len + 1);
+  char *url = xmemdupz(&frag.str[start], (size_t)len + 1);
```

Comment:
might need `assert(len >= 0 && len < MAX_INT32)` for coverity

Reply:
These pre-emptive coverity warning guessing as well as the post-merged coverity warnings adds a lot of friction. I'll see if I have time to set it up coverity CI check for PRs this week, otherwise we might wanna create a dedicated issue to track this problem.